src/auto-evo/simulation/food_source/EnvironmentalFoodSource.cs

```diff
@@ -0,0 +1,47 @@
+﻿public class EnvironmentalFoodSource : IFoodSource
+{
+    private readonly Compound compound;
+    private readonly Compound glucose = SimulationParameters.Instance.GetCompound("glucose");
+
+    private BiomeConditions biomeConditions;
+    private float totalEnvironmentalEnergySource;
+
+    public EnvironmentalFoodSource(Patch patch, string compound, float foodCapacityMultiplier)
+    {
+        biomeConditions = patch.Biome;
+        this.compound = SimulationParameters.Instance.GetCompound(compound);
+        totalEnvironmentalEnergySource = patch.Biome.Compounds[this.compound].Dissolved * foodCapacityMultiplier;
+    }
+
+    public float FitnessScore(Species species)
+    {
+        var microbeSpecies = (MicrobeSpecies)species;
+
+        var energyCreationScore = 0.0f;
+        foreach (var organelle in microbeSpecies.Organelles)
+        {
+            foreach (var process in organelle.Definition.RunnableProcesses)
+            {
+                if (process.Process.Inputs.ContainsKey(compound))
+                {
+                    if (process.Process.Outputs.ContainsKey(glucose))
```

Comment:
Wouldn't it be also nice to check for ATP production here?
Someone might be surprised in the future if this doesn't work similarly in checking the process outputs as the compound food source (might be worth it to make a general helper method to calculate the score from process outputs).

Reply:
I'm happy to take either path if you and @Maxonovien can agree. He argued that the atp calculation should be taken out as dead code.

Reply:
I wouldn't be dead code, right, as it is used by 50% of the implementations.
And if someone makes a mod where an environmental resource makes ATP, they are going to be surprised that auto-evo refuses to evolve that.

Reply:
The mod argument is a satisfying reason for me to keep it ATP calculations

Reply:
Moving the code to a helper would fix this, but since I assume that one of the next steps here would be to differentiate the two (most likely by adding in movement factors to the compound source), I was resisting doing that.

Reply:
You are right in the sense that it is dead code currently. But it might be ran if someone decided to mod photosynthesis to produce ATP instead of glucose. And with the current changes they'll be able to do so only by modifying the .json files, and not touch the actual code. 

This is why it is satisfying enough to me as an argument, for the sake of mod support. The other approach of removing dead code still makes sense in my view, but I for one am open to any decision here, so I have no issue with hh's suggestion.

Reply:
Wouldn't the movement score factors be applied *after* the general computation? So a shared code implementation can be used for that shared part.

Reply:
I don't know. I could see it being implemented that way, I could see it being implemented some other way. Ultimately, this section of code just iterations and counts. I don't foresee the "algorithm" changing, and it would be tricky to make a utility function that is referentially transparent and doesn't make several assumptions about the code calling it. Code reuse is nice, but when you're breaking Single Responsibility Principle to do that, that's when I say you're going too far.

I'd invest more time in this if it was more complicated logic (duplicating the movement logic in the first draft of this PR was dumb and I'm ashamed of that), but this is really just a sum calculation with more complicated accessors.

Reply:
But still, the operation here is the general one "how much energy / useful resource output this process has". According to least astonishment principle that part should not work differently elsewhere. If someone ever adds an environmental energy source that makes ATP directly, they *will* be surprised why auto-evo doesn't want to evolve their cool new organelles.
That's why I want this operation put into a single place, because if there are multiple implementations they may become out of sync (and already has) so they will astonish someone who expects the same operation to work the same in a slightly different context.

Reply:
Something like:
```
compoundUseScore = 0;
                    if (process.Process.Outputs.ContainsKey(glucose))
                    {
                        compoundUseScore += process.Process.Outputs[glucose]
                            / process.Process.Inputs[compound] * Constants.AUTO_EVO_GLUCOSE_USE_SCORE_MULTIPLIER;
                    }

                    if (process.Process.Outputs.ContainsKey(atp))
                    {
                        compoundUseScore += process.Process.Outputs[atp]
                            / process.Process.Inputs[compound] * Constants.AUTO_EVO_ATP_USE_SCORE_MULTIPLIER;
                    }
return compoundUseScore;
```
then calling code has something like: `compoundUseScore += CalculateProcessOutputScore(process)`
it could be a protected method in the abstract base class for all energy sources (which would handle also looking up glucose and ATP).

Reply:
I was overestimating the difficulty in this refactor. It was actually super easy to make this cleanly.

Reply:
One thing that I want to say as slight criticism of your latest commit is that it now looks up the energy sources exponential number of times, as each auto-evo step * each patch * each energy source * each organelle * each process,
is worse than it was before: auto-evo step * each patch * each energy source
so I'd like a way to still keep the once per energy source lookup of the Compound instances.